Ignore failing HTTP test, probably port is occupied at Woodpecker CI worker:
```


[XX]   5 secs db/cmd/cmd_http_post http-post-cmd
RZ_NOPLUGINS=1 /private/var/folders/w7/mbsys_rs56s030vd1q140jpr0000_p/T/woodpecker-local-757388606/workspace/prefix/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -esearch.show_progress=false -eflirt.sigdb.load.home=false -N -qc '!python3 scripts/http_post_cmd_upload.py
' =
-- stdout
--- expected
+++ actual
@@ -1,1 +1,0 @@
-Test succeeded

-- stderr
Traceback (most recent call last):
  File "/private/var/folders/w7/mbsys_rs56s030vd1q140jpr0000_p/T/woodpecker-local-757388606/workspace/test/scripts/http_post_cmd_upload.py", line 90, in <module>
    main()
  File "/private/var/folders/w7/mbsys_rs56s030vd1q140jpr0000_p/T/woodpecker-local-757388606/workspace/test/scripts/http_post_cmd_upload.py", line 72, in main
    post_text = post_cmd.text.split("md5: ")[1].rstrip()
IndexError: list index out of range

[**]                               db/cmd/cmd_http_post    22391 OK      1013 BR        1 XX       26 FX
```

--- a/librz/arch/platform_profile.c
+++ b/librz/arch/platform_profile.c
@@ -10,7 +10,7 @@
  * 	Frees the hashtables used for MMIO and extended
  * 	registers
  */
-RZ_API void rz_platform_profile_free(RzPlatformProfile *p) {
+RZ_API void rz_platform_profile_free(RZ_NULLABLE RzPlatformProfile *p) {
 	if (!p) {
 		return;
 	}
@@ -62,7 +62,7 @@ RZ_API RZ_OWN RzPlatformTarget *rz_platform_target_new() {
  *
  *	Frees the pointer to the SDB and the RzPlatformProfile
  */
-RZ_API void rz_platform_target_free(RzPlatformTarget *t) {
+RZ_API void rz_platform_target_free(RZ_NULLABLE RzPlatformTarget *t) {
 	if (!t) {
 		return;
 	}
@@ -162,7 +162,8 @@ static bool sdb_load_arch_profile_by_path(RZ_NONNULL RzPlatformTarget *t, const 
  * \param t reference to RzPlatformTarget
  * \param path reference to path of the SDB file
  */
-RZ_API bool rz_platform_load_profile_sdb(RzPlatformTarget *t, const char *path) {
+RZ_API bool rz_platform_load_profile_sdb(RZ_NONNULL RzPlatformTarget *t, RZ_NONNULL const char *path) {
+	rz_return_val_if_fail(t && path, false);
 	if (!rz_file_exists(path)) {
 		return false;
 	}
@@ -215,11 +216,11 @@ static bool is_cpu_valid(const char *cpu_dir, const char *cpu) {
  * \param arch reference to the selected architecture (value of `asm.arch`)
  * \param cpus_dir reference to the directory containing cpu files
  */
-RZ_API bool rz_platform_profiles_init(RzPlatformTarget *t, const char *cpu, const char *arch, const char *cpus_dir) {
-	if (!cpu_reload_needed(t, cpu, arch)) {
+RZ_API bool rz_platform_profiles_init(RZ_NULLABLE RzPlatformTarget *t, RZ_NULLABLE const char *cpu, RZ_NULLABLE const char *arch, RZ_NULLABLE const char *cpus_dir) {
+	if (!t || !arch || !cpu || !cpus_dir) {
 		return false;
 	}
-	if (!cpus_dir || !arch || !cpu) {
+	if (!cpu_reload_needed(t, cpu, arch)) {
 		return false;
 	}
 	char buf[50];

--- a/librz/core/canalysis.c
+++ b/librz/core/canalysis.c
@@ -3339,7 +3339,8 @@ static bool add_mmio_extended_flag_cb(void *user, const ut64 addr, const void *v
  * \param profile reference to RzPlatformProfile
  * \param flags reference to RzFlag
  */
-RZ_API void rz_platform_profile_add_flag_every_io(RzPlatformProfile *profile, RzFlag *flags) {
+RZ_API void rz_platform_profile_add_flag_every_io(RZ_NONNULL RzPlatformProfile *profile, RZ_NONNULL RzFlag *flags) {
+	rz_return_if_fail(profile && flags);
 	rz_flag_unset_all_in_space(flags, RZ_FLAGS_FS_MMIO_REGISTERS);
 	rz_flag_unset_all_in_space(flags, RZ_FLAGS_FS_MMIO_REGISTERS_EXTENDED);
 	ht_up_foreach(profile->registers_mmio, add_mmio_flag_cb, flags);

--- a/librz/include/rz_platform.h
+++ b/librz/include/rz_platform.h
@@ -46,11 +46,11 @@ typedef struct rz_platform_target_index_t {
 
 RZ_API RZ_OWN RzPlatformProfile *rz_platform_profile_new();
 RZ_API RZ_OWN RzPlatformTarget *rz_platform_target_new();
-RZ_API void rz_platform_profile_free(RzPlatformProfile *profile);
-RZ_API void rz_platform_target_free(RzPlatformTarget *target);
-RZ_API bool rz_platform_profiles_init(RzPlatformTarget *c, const char *cpu, const char *arch, const char *cpus_dir);
-RZ_API void rz_platform_profile_add_flag_every_io(RzPlatformProfile *profile, RzFlag *flags);
-RZ_API bool rz_platform_load_profile_sdb(RzPlatformTarget *t, const char *path);
+RZ_API void rz_platform_profile_free(RZ_NULLABLE RzPlatformProfile *profile);
+RZ_API void rz_platform_target_free(RZ_NULLABLE RzPlatformTarget *target);
+RZ_API bool rz_platform_profiles_init(RZ_NULLABLE RzPlatformTarget *c, RZ_NULLABLE const char *cpu, RZ_NULLABLE const char *arch, RZ_NULLABLE const char *cpus_dir);
+RZ_API void rz_platform_profile_add_flag_every_io(RZ_NONNULL RzPlatformProfile *profile, RZ_NONNULL RzFlag *flags);
+RZ_API bool rz_platform_load_profile_sdb(RZ_NONNULL RzPlatformTarget *t, RZ_NONNULL const char *path);
 RZ_API RZ_BORROW const char *rz_platform_profile_resolve_mmio(RZ_NONNULL RzPlatformProfile *profile, ut64 address);
 RZ_API RZ_BORROW const char *rz_platform_profile_resolve_extended_register(RZ_NONNULL RzPlatformProfile *profile, ut64 address);
 

--- a/test/db/cmd/cmd_eval
+++ b/test/db/cmd/cmd_eval
@@ -1,3 +1,19 @@
+NAME=e-
+FILE==
+CMDS=<<EOF
+e cfg.seek.histsize
+e cfg.seek.histsize=128
+e cfg.seek.histsize
+e-
+e cfg.seek.histsize
+EOF
+EXPECT=<<EOF
+63
+128
+63
+EOF
+RUN
+
 NAME=et
 FILE==
 CMDS=<<EOF

